prefect_aws/ecs.py

```diff
@@ -531,12 +543,63 @@ async def run(self, task_status: Optional[TaskStatus] = None) -> ECSTaskResult:
         )
 
         return ECSTaskResult(
-            identifier=task_arn,
+            identifier=identifier,
             # If the container does not start the exit code can be null but we must
             # still report a status code. We use a -1 to indicate a special code.
             status_code=status_code or -1,
         )
 
+    @sync_compatible
+    async def kill(self, identifier: str, grace_seconds: int = 30) -> None:
+        """
+        Kill a task running on ECS.
+
+        Args:
+            identifier: A cluster and task arn combination. This should match a value
+                yielded by `ECSTask.run`.
+        """
+        if grace_seconds != 30:
+            self.logger.warning(
+                f"Kill grace period of {grace_seconds}s requested, but AWS does not "
+                "support dynamic grace period configuration so 30s will be used. "
+                "See https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-agent-config.html for configuration of grace periods."  # noqa
+            )
+        cluster, task = parse_task_identifier(identifier)
+        await run_sync_in_worker_thread(self._stop_task, cluster, task)
+
+    def _stop_task(self, cluster: str, task: str) -> None:
+        """
+        Stop a running ECS task.
+        """
+        if self.cluster is not None and cluster != self.cluster:
+            raise InfrastructureNotAvailable(
+                "Cannot stop ECS task: this infrastructure block has access to "
+                f"cluster {self.cluster!r} but the task is running in cluster "
+                f"{cluster!r}."
+            )
+
+        _, ecs_client = self._get_session_and_client()
+        try:
+            ecs_client.stop_task(cluster=cluster, task=task)
+        except Exception as exc:
+            # Raise a special exception if the task does not exist
+            if "ClusterNotFound" in str(exc):
+                raise InfrastructureNotFound(
+                    f"Cannot stop ECS task: the cluster {cluster!r} could not be found."
+                ) from exc
+            if "not find task" in str(exc):
+                raise InfrastructureNotFound(
+                    f"Cannot stop ECS task: the task {task!r} could not be found in "
+                    f"cluster {cluster!r}."
+                ) from exc
+            if "no registered tasks" in str(exc):
+                raise InfrastructureNotFound(
+                    f"Cannot stop ECS task: the cluster {cluster!r} has no tasks."
+                ) from exc
```

Comment:
```suggestion
            exc_msg = str(exc)
            if "ClusterNotFound" in exc_msg:
                raise InfrastructureNotFound(
                    f"Cannot stop ECS task: the cluster {cluster!r} could not be found."
                ) from exc
            if "not find task" in exc_msg:
                raise InfrastructureNotFound(
                    f"Cannot stop ECS task: the task {task!r} could not be found in "
                    f"cluster {cluster!r}."
                ) from exc
            if "no registered tasks" in exc_msg:
                raise InfrastructureNotFound(
                    f"Cannot stop ECS task: the cluster {cluster!r} has no tasks."
                ) from exc
```

Reply:
I'm not sure this is a particularly meaningful change for performance, is this for readability?

Reply:
I personally like to define a variable if used multiple times, but I don't feel strongly about it.